### PR TITLE
Users manage page

### PR DESCRIPTION
**Pull request by aleksaalfir**
A page where the administrator can see all users and where he can manage them...

### CODE REVIEW

File: src/components/Users/User/User.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import UserModel from 'model/UserModel';
+import classes from './User.module.scss';
+
+interface UserProps {
+  user : UserModel;
+  editUser: (id: string | number, user: UserModel) => void;
+  deleteUser: (id: string | number) => void;
+}
+
+const User: React.FC<UserProps> = ({user, editUser, deleteUser}) => {
```
Comment (petarnqode) [inline src/components/Users/User/User.tsx]: Can we dest object agin, exp. { user :{ title, startRentDate, endRentDate }, editUser, deleteUser }

**Review by petarnqode — CHANGES_REQUESTED** [review]

File: src/components/Users/User/User.module.scss [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    display: grid;
+    grid-template-columns: repeat(3, 1fr);
+
+    padding: $space;
+
+    border-radius: $space-l;
+
+    text-align: center;
+
+    color: $color-white;
```
Comment (vkotarlic) [inline src/components/Users/User/User.module.scss]: `text-align` and `color` should be grouped together

File: src/components/Users/User/User.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+  const [toggleDetails, setToggleDetails] = useState<boolean>(false);
+  const [userValues, setUserValues] = useState<UserModel>(user);
+
+  const changeUserValueHandler = (value : string,  prop?: string) =>{
```
Comment (vkotarlic) [inline src/components/Users/User/User.tsx]: Please add space after `=>`

File: src/components/Users/User/User.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        <div>{userValues.id}</div>
+        <div>{userValues.email}</div>
+        <div>{userValues.phoneNumber}</div>
+      </div>
+      {toggleDetails ? 
+            <div className={classes['c-user__all-details']}>
+            <div className={classes['c-user__input-holder']}>
+              <Input
+                type={'text'}
+                placeholder={'first name'}
```
Comment (vkotarlic) [inline src/components/Users/User/User.tsx]: Could you please format this placeholder to be `First name` or `Please enter first name`

File: src/components/Users/User/User.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+                id={'firstName'}
+                name={'firstName'}
+                inputValue={userValues.firstName}
+                setValue={changeUserValueHandler}
+              />
+              <Input
+                type={'text'}
+                placeholder={'last name'}
```
Comment (vkotarlic) [inline src/components/Users/User/User.tsx]: Could you adjust this placeholder as it is suggested in comment above

File: src/components/Users/User/User.tsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+                type={'text'}
+                placeholder={'email'}
+                id={'email'}
+                name={'email'}
+                inputValue={userValues.email}
+                setValue={changeUserValueHandler}
+              />
+              <Input
+                type={'text'}
+                placeholder={'phone number'}
```
Comment (vkotarlic) [inline src/components/Users/User/User.tsx]: Could you adjust this placeholder as it is suggested in comment above

File: src/components/Users/User/User.tsx [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+                name={'phoneNumber'}
+                inputValue={userValues.phoneNumber}
+                setValue={changeUserValueHandler}
+              />
+            </div>
+            <div className={classes['c-user__input-holder']}>
+              <div className={classes['c-user__last']}>
+                <Input
+                  type={'address'}
+                  placeholder={'address'}
```
Comment (vkotarlic) [inline src/components/Users/User/User.tsx]: Could you adjust this placeholder as it is suggested in comment above

File: src/components/Users/User/User.module.scss [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    padding: $space;
+    margin-bottom: $space;
+  }
+
+  &__input-holder {
+    display: flex;
+    align-items: center;
+  }
+
+  &__last {
```
Comment (vkotarlic) [inline src/components/Users/User/User.module.scss]: What is `__last` i think this class should be named something more meaningful

File: src/components/Users/User/User.tsx [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+                type={'text'}
+                placeholder={'phone number'}
+                id={'phoneNumber'}
+                name={'phoneNumber'}
+                inputValue={userValues.phoneNumber}
+                setValue={changeUserValueHandler}
+              />
+            </div>
+            <div className={classes['c-user__input-holder']}>
+              <div className={classes['c-user__last']}>
```
Comment (vkotarlic) [inline src/components/Users/User/User.tsx]: Please check comment in scss about `__last`

File: src/components/Users/Users.tsx [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+  };
+
+  const editUser = (id: number | string, user: UserModel): void => {
+    axios
+      .put(`/user/${id}`, user)
+      .then(() => {
+        toastService.toastSuccess(`User with ID: ${id} updated!`);
+      })
+      .catch((err) => {
+        toastService.toastError(`${err.response.data.message}`);
```
Comment (vkotarlic) [inline src/components/Users/Users.tsx]: Please add some error message for this don't use backend one

File: src/components/Users/Users.tsx [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+  };
+
+  const deleteUser = (id: number | string): void => {
+    axios
+      .delete(`/user/${id}`)
+      .then(() => {
+        toastService.toastSuccess(`User with ID: ${id} deleted!`);
+      })
+      .catch((err) => {
+        toastService.toastError(`${err.response.data.message}`);
```
Comment (vkotarlic) [inline src/components/Users/Users.tsx]: Please add some error message for this don't use backend one

File: src/pages/Login/Login.tsx [hunk 12]
```diff
@@ -22,6 +22,7 @@ const Login: React.FC = () => {
     axios
       .post(`/authenticate`, loginData)
       .then((response) => {
+        console.log("aaaa");
```
Comment (vkotarlic) [inline src/pages/Login/Login.tsx]: Please remove this console.log 😆

File: src/services/toastService.tsx [hunk 13]
```diff
@@ -0,0 +1,37 @@
+import { toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+
+// IMPORT <ToastContainer/> in component where you want to use toastService
```
Comment (vkotarlic) [inline src/services/toastService.tsx]: Please remove this comment it doesn't have any purpose

File: src/services/toastService.tsx [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+        position: "top-right",
+        autoClose: 3000,
+        hideProgressBar: false,
+        closeOnClick: true,
+        pauseOnHover: true,
+        draggable: true,
+        progress: undefined,
+        theme: "colored",
+        });
+}
```
Comment (vkotarlic) [inline src/services/toastService.tsx]: Please format this code with prettier

**Review by vkotarlic — CHANGES_REQUESTED** [review]

**Pull request merged**